tf_quant_finance/models/heston/approximations/heston_test.py

```diff
@@ -0,0 +1,157 @@
+
+    return (a + b) / (1.0 + u * u)
+
+  k = np.log(strike / forward)
+
+  integral = integrate.quad(
+    lambda u: integrand_function(u, k),
+    0,
+    float("inf")
+  )[0]
+
+  undiscount_call_price = (forward - strike * (0.5 + 1 / math.pi * integral))
+  undiscount_put_price = undiscount_call_price - forward + strike
+  call_price = undiscount_call_price * discount_factor
+  put_price = undiscount_put_price * discount_factor
+
+  return call_price, put_price
+
+
+@test_util.run_all_in_graph_and_eager_modes
+class HestonPriceTest(parameterized.TestCase, tf.test.TestCase):
+  """Tests for Heston Price method."""
+
+  # TODO: Need to test single precision (test wouldn't pass with simpsons
+  #  integration)
+  @parameterized.named_parameters(
+    {
+      'testcase_name': 'DoublePrecision',
+      'dtype': np.float64
+    })
+  def test_heston_price(self, dtype):
+    kappas = np.asarray([0.1, 10.0], dtype=dtype)
+    thetas = np.asarray([0.1, 0.5], dtype=dtype)
+    v0s = np.asarray([0.1, 0.5], dtype=dtype)
+    forwards = np.asarray([10.0], dtype=dtype)
+    sigmas = np.asarray([1.0], dtype=dtype)
+    strikes = np.asarray([9.7, 10.0, 10.3], dtype=dtype)
+    expiries = np.asarray([1.0], dtype=dtype)
+    discount_factors = np.asarray([0.99], dtype=dtype)
+
+    # TODO: test rhos = [-0.5, 0, 0.5]
+    #  (test wouldn't pass with simpsons integration)
+    rhos = np.asarray([0], dtype=dtype)
+
+    for kappa in kappas:
+      for theta in thetas:
+        for sigma in sigmas:
+          for rho in rhos:
+            for v0 in v0s:
+              for forward in forwards:
+                for expiriy in expiries:
+                  for strike in strikes:
+                    for discount_factor in discount_factors:
+                      tff_prices = self.evaluate(
+                        tff.models.heston.approximations.eu_option_price(
+                          kappas=np.asarray([kappa]),
+                          thetas=np.asarray([theta]),
+                          sigmas=np.asarray([sigma]),
+                          rhos=np.asarray([rho]),
+                          variances=np.asarray([v0]),
+                          forwards=np.asarray([forward]),
+                          expiries=np.asarray([expiriy]),
+                          strikes=np.asarray([strike]),
+                          discount_factors=np.asarray([discount_factor]),
+                          is_call_options=np.asarray([True, False], dtype=np.bool)
+                        ))
+
+                      params = {
+                        "kappa": kappa,
+                        "theta": theta,
+                        "sigma": sigma,
+                        "rho": rho,
+                        "v0": v0,
+                        "forward": forward,
+                        "expiry": expiriy,
+                        "strike": strike,
+                        "discount_factor": discount_factor,
+                      }
```

Comment:
This is a bit dense. I have addressed our previous conversation. Could you please remove `get_heston_prices` and just hardcode the exact answer?

As for me, computing `tff.models.heston.approximations.eu_option_price(kappas=kappas, ..)` is better (making sure that the inputs are broadcastable). It also checks that the vectorization is supported.